src/vec_map.rs

```diff
@@ -0,0 +1,50 @@
+use serde::{ser::SerializeMap, Serialize, Serializer};
+use std::collections::HashMap;
+
+/// Use a sorted Vec to store key-value pairs.
+/// Useful as a replacement of HashMap/BTreeMap where:
+/// - There is no need for fast access of values by keys.
+/// - The keys need to be sorted to be use as partition key
+/// for PartitionBatchSink.
+///
+/// Ideal to store labels map in various sinks.
+#[derive(Debug, Clone)]
+pub struct VecMap<V> {
```

Comment:
I'm not sure I like the naming of these two structures, but the idea is reasonable. They are named `Map` but don't support any of the normal map accessors. This seems like more of a `SortedVec`.